### PR TITLE
#3: Add deploy command

### PR DESCRIPTION
**Pull request by zachelrath**
Addresses #8 - Add deploy command

### CODE REVIEW

File: cmd/retrieve.go [hunk 1]
```diff
@@ -50,16 +52,27 @@ var retrieveCmd = &cobra.Command{
 
 		fmt.Println("Retrieving metadata...")
 
-		//query the API for all Skuid metadata of every type
-		result, err := api.Connection.MakeRequest(http.MethodPost, "/metadata/retrieve", retrieveRequest)
+		var body io.Reader
 
-		fmt.Println("Successfully retrieved metadata!")
+		if retrieveRequest != nil {
```
Comment (mrhwick) [inline cmd/retrieve.go]: Is there a case where `retrieveRequest` would equal nil? Empty struct?

File: cmd/deploy.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+				fmt.Println("Adding file to ZIP:", archivePath)
+			}
+		}
+
+		err = zip.ArchiveFile(targetDir, outFilePath, progress)
+		if err != nil {
+		    panic(err)
+		}
+
+		reader, err := os.Open(outFilePath)
```
Comment (zachelrath) [inline cmd/deploy.go]: done

File: cmd/retrieve.go (hunk 1 above)
Comment (zachelrath) [inline cmd/retrieve.go]: done

File: cmd/deploy.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		}
+
+		if verbose {
+			fmt.Println("Deploying site from", targetDirFriendly)
+		}
+
+		// Create a temporary directory into which to put our ZIP
+		tmpDir, err := ioutil.TempDir("", "skuid-deploy")
+		if err != nil {
+			panic(err)
```
Comment (ethanfrogers) [inline cmd/deploy.go]: We probably shouldn't panic here as the user will have no idea what's _actually_ wrong. Maybe log a human parsable error, and exit early.

File: cmd/deploy.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+		progress := func(archivePath string) {
+			if verbose {
+				fmt.Println("Adding file to ZIP:", archivePath)
+			}
+		}
+
+		err = zip.ArchiveFile(targetDir, outFilePath, progress)
+		if err != nil {
+			panic(err)
```
Comment (ethanfrogers) [inline cmd/deploy.go]: Same as above.

File: cmd/deploy.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		}
+
+		reader, err := os.Open(outFilePath)
+
+		defer reader.Close()
+
+		if err != nil {
+			log.Fatal(err)
```
Comment (ethanfrogers) [inline cmd/deploy.go]: Early exit here so as to not continue execution in the event of a failure.

**Pull request merged**